tools/lsp/main.rs

```diff
@@ -282,6 +284,7 @@ fn main_loop(connection: Connection, init_param: InitializeParams) -> Result<()>
                         OutgoingRequest::Done(_) => {
                             return Err("Response to unknown request".into())
                         }
+                        OutgoingRequest::Start => { /* nothing to do */ }
```

Comment:
Looks like the request is going to be leaking in that case, isn't it?

Reply:
Ah, no, on second thought i guess that's fine because we set the request to be Done right after that, and i guess we'lll eventually poll this future anyway even if we don't wake it explicitly